api/src/main/java/com/datatorrent/api/Operator.java

```diff
@@ -172,6 +172,51 @@ public String toString()
 
   }
 
+  public interface ProxyPort<T> extends Port
+  {
+    void set(T port);
+
+    T get();
+  }
+
+  public final class ProxyInputPort<T> extends DefaultInputPort<T> implements ProxyPort<InputPort<T>>
```

Comment:
Agreed. If someone tries to use these proxy ports like actual ports, the functionality must be delegated to the actual ports. Will add this fix.